No OASFactoryResolver implementation found! when running native application

### Describe the bug

Hello,

I'm running into the following error when trying to start my natively compiled application docker container:

```
Exception in thread "main" java.lang.ExceptionInInitializerError
        at java.lang.Class.ensureInitialized(DynamicHub.java:510)
        at java.lang.Class.newInstance(DynamicHub.java:874)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 6 more
Caused by: java.lang.IllegalStateException: No OASFactoryResolver implementation found!
        at org.eclipse.microprofile.openapi.spi.OASFactoryResolver.instance(OASFactoryResolver.java:77)
        at io.quarkus.smallrye.openapi.runtime.OpenApiRecorder.classLoaderHack(OpenApiRecorder.java:80)
        at io.quarkus.deployment.steps.SmallRyeOpenApiProcessor$classLoaderHack1558305429.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.SmallRyeOpenApiProcessor$classLoaderHack1558305429.deploy(Unknown Source)
        ... 7 more

```

Is there something I'm missing? Should OpenAPI be disabled in production?

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### GraalVM version (if different from Java)

GraalVM 22.0.0.2 Java 17 CE

### Quarkus version or git rev

2.7.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 

### Additional information

_No response_

--- a/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
+++ b/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
@@ -31,6 +31,7 @@ import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
 import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
 import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
 import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.eclipse.microprofile.openapi.spi.OASFactoryResolver;
 import org.jboss.jandex.AnnotationInstance;
 import org.jboss.jandex.AnnotationTarget;
 import org.jboss.jandex.AnnotationValue;
@@ -156,9 +157,10 @@ public class SmallRyeOpenApiProcessor {
     }
 
     @BuildStep
-    void registerNativeImageResources(BuildProducer<ServiceProviderBuildItem> serviceProvider) throws IOException {
+    void registerNativeImageResources(BuildProducer<ServiceProviderBuildItem> serviceProvider) {
         // To map from smallrye and mp config to quarkus
         serviceProvider.produce(ServiceProviderBuildItem.allProvidersFromClassPath(OpenApiConfigMapping.class.getName()));
+        serviceProvider.produce(ServiceProviderBuildItem.allProvidersFromClassPath(OASFactoryResolver.class.getName()));
     }
 
     @BuildStep

